src/components/List/List.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import bem from 'bem';
+import styles from './List.scss';
+
+const { block } = bem({
+    name: 'List',
+    classnames: styles,
+    propsToMods: ['isDividedList', 'Component']
+});
+
+const List = props => {
+    const { children, Component, isDividedList, ...rest } = props;
+    return (
+        <Component {...rest} {...block(props)}>
+            {React.Children.map(children, child => React.cloneElement(child, { isDividedList }))}
+        </Component>
+    );
+};
+
+List.displayName = 'List';
+
+List.propTypes = {
+    /** List Items */
+    children: PropTypes.node,
+    /** The HTML or React component to be used to render the list  */
+    Component: PropTypes.string,
+    /** Adds dividing lines between the list items */
+    isDividedList: PropTypes.bool
+};
+
+List.defaultProps = {
+    children: null,
+    Component: 'ul',
+    isDividedList: false
```

Comment:
I would change this to just `isDivided`. To render the divider, you could then do e.g. this:

```
{React.Children.map(children, child => React.cloneElement(child, { ...elem('item', props) }))}
...
.List__item + .List__item {
   border-top: 1px solid color;
}
```

(note that I'm adding a element class to the cloned element)

Reply:
This is just an other approach to what I did. Is there a specific advantage you see in this one as opposed to the current one?

Reply:
this is just an other approach to what is currently implemented. Is there a specific reason you think would be better?

Reply:
In a nutshell I just think that supporting division on Item level is not really necessary, and `isDivided` on list level makes more sense to me in terms of naming because it's on List level anyway (so the name `isDividedList` is a bit double). In terms of CSS I personally think using adjacent sibling selectors is a bit more readable (and that's how I think it's typically done) over using multiple pseudo-selectors, but that's more a matter of preference. And as mentioned before I would consider always using one particular list type (`ul` would be fine for the current use case?) for now instead of making that configurable, to keep it simpler. If necessary we can make it configurable later on, right?

Reply:
I don't see much sense in implementing such an abstract and generic `List` component. If we really need to support ordered and unordered lists (which I doubt), might be make sense to create two different components: `OrderedList` and `UnorderedList`. Or at least the interface of the component should be more semantic, like  `<List  type="ordered|unordered" />`.

One more thought on this. I think it's a part of responsibility of this component to render tags for items. In this case we can do 2 things:

1. control that `<ul>` tag contains only allowed `<li>` tags inside
2. Don't need to clone children elements and passing `isDividedList`.

```
return (
  <ul {...rest} {...block(props)}>
     {React.Children.map(children, child => (
        <li {...elem('item', props)}>
          {child}
        </li>
      ) 
    ))}
  </Component>
);
```

Reply:
changed to use `isDivided` on list only.